File: py/tcp_server.py
import socket

def server():
    host = socket.gethostname()
    port = 8080

    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.bind((host, port))

    s.listen(1)
    c, addr = s.accept()

    print("Connection from: " + str(addr))
    try:
        while True:
            data = c.recv(1024).decode('utf-8')
            if not data:
              break
            print('peer: ' + data)
            msg = input('==> ')
            c.send(msg.encode('utf-8'))
    except KeyboardInterrupt:
        c.close()
        exit(0)


if __name__ == '__main__':
    server()

File: test-files/tcp-server.js
const net = require('net')

var server = net.createServer();

server.on('connection', function(c) {
    console.log("New Connection!");
    c.write("Hello!");
});

server.on('data', function(data) {
	console.log('Received: ' + data);
});

server.on('close', function(data) {
    console.log('');
});

// server.listen(1337, '127.0.0.1');
server.listen(1337, '169.254.146.198');
